code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -770,19 +790,636 @@ nobiliumsuppression = INFINITY
+	energy_released += consumed_fuel * 800
+	if(energy_released)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+	return REACTING
+
+/datum/gas_reaction/cyrion_b_crystal_formation
+	priority = 19
+	name = "cyrion_b crystal formation"
+	id = "cyrion_b_crystal_formation"
+
+/datum/gas_reaction/cyrion_b_crystal_formation/init_reqs()
+	min_requirements = list(
+		/datum/gas/plasma = 50,
+		/datum/gas/cyrion_b = 10,
+		"TEMP" = 270,
+		"MAX_TEMP" = 280
+	)
+
+/datum/gas_reaction/cyrion_b_crystal_formation/react(datum/gas_mixture/air, datum/holder)
+	var/energy_released = 0
+	var/old_heat_capacity = air.heat_capacity()
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/turf/open/location = isturf(holder) ? holder : null
+	var/consumed_fuel = 0
+	consumed_fuel = min(cached_gases[/datum/gas/cyrion_b][MOLES] * 0.1, 20 * (temperature * 0.02))
+	if ((cached_gases[/datum/gas/cyrion_b][MOLES] - consumed_fuel * 0.05 < 0 ) || (cached_gases[/datum/gas/plasma][MOLES] - consumed_fuel * 5 < 0)) //Shouldn't produce gas from nothing.
+		return NO_REACTION
+	cached_gases[/datum/gas/plasma][MOLES] -= consumed_fuel * 5
+	cached_gases[/datum/gas/cyrion_b][MOLES] -= consumed_fuel * 0.05
+	if(prob(10 * consumed_fuel))
+		new /obj/item/grenade/gas_crystal/cyrion_b_crystal(location)
+	energy_released += consumed_fuel * 800
+	if(energy_released)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+	return REACTING
+
+/datum/gas_reaction/proto_nitrate_plasma_response
+	priority = 20
+	name = "Proto Nitrate plasma response"
+	id = "proto_nitrate_plasma_response"
+
+/datum/gas_reaction/proto_nitrate_plasma_response/init_reqs()
+	min_requirements = list(
+		/datum/gas/proto_nitrate = MINIMUM_MOLE_COUNT,
+		/datum/gas/plasma = MINIMUM_MOLE_COUNT,
+		"TEMP" = 250,
+		"MAX_TEMP" = 300
+	)
+
+/datum/gas_reaction/proto_nitrate_plasma_response/react(datum/gas_mixture/air, datum/holder)
+	var/energy_released = 0
+	var/old_heat_capacity = air.heat_capacity()
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/turf/open/location = isturf(holder) ? holder : null
+	if(cached_gases[/datum/gas/plasma][MOLES] < 10)
+		var produced_amount = min(5, cached_gases[/datum/gas/plasma][MOLES], cached_gases[/datum/gas/proto_nitrate][MOLES])
+		if(cached_gases[/datum/gas/plasma][MOLES] - produced_amount < 0 || cached_gases[/datum/gas/proto_nitrate][MOLES] - produced_amount * 0.1 < 0)
+			return NO_REACTION
+		cached_gases[/datum/gas/plasma][MOLES] -= produced_amount
+		cached_gases[/datum/gas/proto_nitrate][MOLES] -= produced_amount * 0.1
+		energy_released -= produced_amount * 1500
+		if(prob(produced_amount * 15))
+			new/obj/item/stack/sheet/mineral/plasma(location)
```

Comment:
this'll make crystals in nullspace right? that seems wrong. Why are ya doing it this way?

Reply:
changing all the obj making reactions to open air only